src/pool-bin/BinPoolManagerOwner.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2024 PancakeSwap
+pragma solidity 0.8.26;
+
+import {Currency} from "../types/Currency.sol";
+import {IProtocolFeeController} from "../interfaces/IProtocolFeeController.sol";
+import {PausableRole} from "../base/PausableRole.sol";
+import {IBinPoolManager} from "./interfaces/IBinPoolManager.sol";
+import {IPoolManagerOwner} from "../interfaces/IPoolManagerOwner.sol";
+
+/**
+ * @dev This contract is the owner of the BinPoolManager contract
+ *
+ * A seperate owner contract is used to handle some functionality so as to reduce the contract size
+ * of PoolManager. This allow a higher optimizer run, reducing the gas cost for other poolManager functions
+ */
+contract BinPoolManagerOwner is IPoolManagerOwner, PausableRole {
+    /// @notice Error thrown when owner set min share too small
+    error MinShareTooSmall();
+
+    IBinPoolManager public immutable poolManager;
+
+    constructor(IBinPoolManager _poolManager) {
+        poolManager = _poolManager;
+    }
+
+    /// @inheritdoc IPoolManagerOwner
+    function pausePoolManager() external override onlyPausableRoleOrOwner {
```

Comment:
can we revokePausableRole  when this role paused once ?
my suggestion , one role can only pause once.

Reply:
whats the scenario or context where you see we need to revoke always? 

if the EOA is compromised and pause(), we'll be alerted on a `pause()` event and check why, will then manually revoke it after investigation.

Reply:
ok , got it